Managers/UInputMan.cpp

```diff
@@ -981,4 +1012,74 @@ namespace RTE {
 			}
 		}
 	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#ifdef __unix__
+	void UInputMan::HandleAllegroMouseInput() {
+		if (_xwin.display == 0)
+			return;
+
+		int nEvents = XPending(_xwin.display);
+		std::vector<XEvent> events;
+		events.reserve(nEvents + 10);
+		for (int i = 0; i < XPending(_xwin.display); ++i) {
+			XEvent e;
+			XNextEvent(_xwin.display, &e);
+			events.push_back(e);
+		}
+
+		static int mousePrevX = 0;
+		static int mousePrevY = 0;
+		mousePrevX = mouse_x;
+		mousePrevY = mouse_y;
+		int dx = 0;
+		int dy = 0;
+
+		int halfResX = _xwin.window_width / 2;
+		int halfResY = _xwin.window_height / 2;
+		for (auto event = events.rbegin(); event < events.rend(); ++event) {
+			switch (event->type) {
+				case MotionNotify: {
+					dx = event->xmotion.x - mousePrevX;
+					dy = event->xmotion.y - mousePrevY;
+					_xwin_mouse_interrupt(dx, dy, 0, 0, mouse_b);
+					_mouse_x = mousePrevX = !g_UInputMan.m_TrapMousePos ? event->xmotion.x : halfResX;
+					_mouse_y = mousePrevY = !g_UInputMan.m_TrapMousePos ? event->xmotion.y : halfResY;
+
+					if (g_UInputMan.m_TrapMousePos && (dx != 0 || dy != 0)) {
+						XWarpPointer(_xwin.display, _xwin.window, _xwin.window, 0, 0, 0, 0, halfResX, halfResY);
+					}
+				} break;
```

Comment:
Can you move this break into the `{}`s safely/without changing the code's meaning?? If so, please do.
Also, no need for the empty line before the default case